modules/porous_flow/test/tests/dirackernels/hfrompps.i

```diff
@@ -0,0 +1,280 @@
+  [../]
+
+  [./T_in_fn]
+    type = PiecewiseLinear
+    xy_data = '
+      0    400
+      500  450'
+  [../]
+[]
+
+[BCs]
+  [./P_top]
+    type = DirichletBC
+    preset = true
+    variable = pressure
+    boundary = top
+    value = 1e6
+  [../]
+
+  [./T_top]
+    type = DirichletBC
+    preset = true
+    variable = temperature
+    boundary = top
+    value = 400
+  [../]
+
+  [./T_bot]
+    type = DirichletBC
+    preset = true
+    variable = temperature
+    boundary = bottom
+    value = 400
+  [../]
+[]
+
+[Modules]
+  [./FluidProperties]
+    [./water]
+      type = Water97FluidProperties
+    [../]
+  [../]
+[]
+
+[Materials]
+  [./temperature]
+    type = PorousFlowTemperature
+    temperature = temperature
+  [../]
+
+  [./temperature_nodal]
+    type = PorousFlowTemperature
+    at_nodes = true
+    temperature = temperature
+  [../]
+
+  [./ppss]
+    type = PorousFlow1PhaseP
+    at_nodes = true
+    porepressure = pressure
+    capillary_pressure = pc
+  [../]
+
+  [./ppss_qp]
+    type = PorousFlow1PhaseP
+    porepressure = pressure
+    capillary_pressure = pc
+  [../]
+
+  [./massfrac]
+    type = PorousFlowMassFraction
+    at_nodes = true
+  [../]
+
+  [./fluid_props]
+    type = PorousFlowSingleComponentFluid
+    fp = water
+    phase = 0
+    at_nodes = true
+  [../]
+
+  [./fluid_props_qp]
+    type = PorousFlowSingleComponentFluid
+    phase = 0
+    fp = water
+  [../]
+
+  [./relperm]
+    type = PorousFlowRelativePermeabilityCorey
+    at_nodes = true
+    n = 1
+    phase = 0
+  [../]
+
+  [./relperm_qp]
+    type = PorousFlowRelativePermeabilityCorey
+    n = 1
+    phase = 0
+  [../]
+
+  [./fp_mat]
+    type = FluidPropertiesMaterialPT
+    pressure = pressure
+    temperature = temperature
+    fp = water
+  [../]
+
+  [./rock_heat]
+    type = PorousFlowMatrixInternalEnergy
+    specific_heat_capacity = 830.0
+    density = 2750
+  [../]
+
+  [./thermal_conductivity]
+    type = PorousFlowThermalConductivityIdeal
+    dry_thermal_conductivity = '2.5 0 0  0 2.5 0  0 0 2.5'
+  [../]
+
+  [./permeability]
+    type = PorousFlowPermeabilityConst
+    permeability = '1.0E-15 0 0  0 1.0E-15 0  0 0 1.0E-14'
+  [../]
+
+  [./porosity_qp]
+    type = PorousFlowPorosityConst
+    porosity = 0.1
+  [../]
+[]
+
+[DiracKernels]
+  [./source]
+    type = PorousFlowPointSourceFromPostprocessor
+    variable = pressure
+    mass_flux = mass_flux_in
+    point = '0.5 0.5 0'
+  [../]
+  [./source_h]
+    type = PorousFlowPointEnthalpySourceFromPostprocessor
+    variable = temperature
+    mass_flux = mass_flux_in
+    point = '0.5 0.5 0'
+    T_in = T_in
+    pressure = pressure
+    fp = water
+  [../]
+[]
+
+[Preconditioning]
+  [./preferred]
+    type = SMP
+    full = true
+    petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
+    petsc_options_value = ' lu       mumps'
+  [../]
+[]
+
+[Postprocessors]
+  [./total_mass]
+    type = PorousFlowFluidMass
+    execute_on = 'initial timestep_end'
+  [../]
```

Comment:
You should also have a `PorousFlowHeatEnergy` Postprocessor so that your gold file demonstrates the correct increasing value of fluid heat energy